File: seeds/esp32-tlora-pager/tools/test_settings_policy.py
#!/usr/bin/env python3
from pathlib import Path

root = Path(__file__).resolve().parents[1]
main = (root / "src/main.cpp").read_text(encoding="utf-8")
kb = (root / "src/hw_kb.cpp").read_text(encoding="utf-8")
ui = (root / "src/hw_ui.cpp").read_text(encoding="utf-8")

assert '#define SETTINGS_PATH     "/settings.json"' in main
assert "write_spiffs_file_atomic(SETTINGS_PATH, SETTINGS_TMP, body);" in main
assert 'doc["silent"].is<bool>()' in main
assert "settings_silent = !settings_silent;" in main
assert "if (!settings_silent) hw_sound_notify(lvl);" in main
assert "hw_haptic_notify(lvl);" in main
assert "hw_kb_take_silent_toggle()" in main
assert "pressed && alt_held && row == 1 && col == 1" in kb
assert '"SILENT %s", silent ? "ON" : "OFF"' in ui
assert 'snprintf(row2, sizeof(row2), "SILENT");' in main

print("settings policy tests: OK")

File: seeds/esp32-tlora-pager/tools/test_settings_policy.cpp
#include <assert.h>
#include <stdint.h>
#include <stdio.h>
#include "../src/settings_policy.h"
int main(void) {
    assert(settings_autolock_next(0) == 30);
    assert(settings_autolock_next(30) == 60);
    assert(settings_autolock_next(60) == 300);
    assert(settings_autolock_next(300) == 0);
    assert(!settings_autolock_due(30, 29999, false));
    assert(settings_autolock_due(30, 30000, false));
    assert(!settings_autolock_due(30, 30000, true));
    assert(!settings_autolock_due(0, UINT32_MAX, false));
    puts("settings autolock tests: OK");
}
